Set Up Docker

KramarenkoVladyslav opened the pull request.

mehalyna commented on djangoProject/Dockerfile:
```diff
@@ -0,0 +1,22 @@
+FROM python:3.11.4-slim
```
Pin the specific version to avoid accidental upgrades that may cause compatibility issues:
```
FROM python:3.11.4-slim@sha256:<specific-digest>
```

mehalyna commented on djangoProject/Dockerfile:
```diff
@@ -0,0 +1,22 @@
+COPY requirements.txt /app/
+
+# Install dependencies
+RUN pip install --no-cache-dir -r requirements.txt
```
Use a virtual environment to isolate dependencies:
```
RUN python -m venv /venv && \
    . /venv/bin/activate && \
    pip install --no-cache-dir -r requirements.txt
ENV PATH="/venv/bin:$PATH"
```
Use a non-root user for installation to improve security:
```
RUN adduser --disabled-password appuser
USER appuser
```

mehalyna commented on djangoProject/Dockerfile:
```diff
@@ -0,0 +1,22 @@
+RUN pip install --no-cache-dir -r requirements.txt
+
+# Copy the rest of the project files into the container
+COPY . /app/
```
Avoid copying sensitive files (e.g., `.env`, `.git`, or cache files).
Use a `.dockerignore` file to exclude files that aren’t required in production.

mehalyna reviewed (changes requested):
Exclude all bin files from PR. pls

VovaD199 commented on djangoProject/djangoProject/settings.py:
```diff
@@ -79,13 +79,20 @@
 # https://docs.djangoproject.com/en/5.1/ref/settings/#databases
 
+import os
+
```
This is a duplicate of line 14.

KramarenkoVladyslav commented:
I have completed all the requested changes. Apologies for including the bin files in the previous commit; I’ve now removed them and updated the `.gitignore` file to avoid this issue in the future.
